client/components/data/query-themes/index.jsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import shallowEqual from 'react-pure-render/shallowEqual';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isJetpackSite } from 'state/sites/selectors';
+import { requestThemes } from 'state/themes/actions';
+import { isRequestingThemesForQuery } from 'state/themes/selectors';
+
+class QueryThemes extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		query: PropTypes.object,
```

Comment:
Good point. With the changes from bb91403, `siteId`'s `propTypes` have changed quite a bit, but they can made required (which I'm doing in 2d8b37b). OTOH, `query` can be falsey -- the action (and the `wpcom.undocumented()` call) will just fall back to "no query", i.e. all results, unfiltered.